### DISCUSSION TITLE
How to see/set subscriber session AMBR details on R1.4 AGW/NMS

### DISCUSSION THREAD

**Discussion by pkapatsos-blinq** [magma/magma#5816] [Q&A] [answered]
We are trying to see what QOS or UL/DL AMBR setting a subscriber is getting vs what is set at the APNs .
Subscriber_cli.py shows what is being sent from the NMS , but our CPE indicates and PCAP of ICS showing the default data plan numbers is being sent/set for the UE  not APN values. 
APN view on NMS
![image](https://user-images.githubusercontent.com/61711855/113019124-428e5480-914f-11eb-9ddf-e3e434070e7f.png)

The subscriber_cli.py view on AGW, 
![image](https://user-images.githubusercontent.com/61711855/113019684-dbbd6b00-914f-11eb-87b1-bc849d98a793.png)

Subscriber attached view on NMS
![image](https://user-images.githubusercontent.com/61711855/113021179-520e9d00-9151-11eb-914a-2536b1209f53.png)

JSON view of sub showing value for subscribedUEAmbr
![image](https://user-images.githubusercontent.com/61711855/113021482-9f8b0a00-9151-11eb-8360-95d88233d9d9.png)

We are seeing the default DATA PLAN being set for CPEs no matter what APN rates you apply, how can we change please.
![image](https://user-images.githubusercontent.com/61711855/113019436-9bf68380-914f-11eb-9a3e-feccc4c47321.png)

How can add new data plans or adjust/modify the default or why are the APN AMBR not being set , without going into UE metering setups?  
Also how to view the session details on a subscriber of  what setting is being used on the AGWs as  
session_manager_cli.py is failing .
![image](https://user-images.githubusercontent.com/61711855/113022359-846cca00-9152-11eb-80fe-13ee66228ff6.png)

Thank you .

**Comment by ulaskozat** [comment]
APN AMBR values are currently hardwired in communication towards UE (they are enforced on the AGW side). Subscribers can also have a data plan which sets UE-AMBR.

**Reply by pkapatsos-blinq** [reply]
@karthiksubraveti  adding from slack discussion 
@karthik subraveti we used to have a data plan configuration in old NMS, I cannot see it on NMS dashboard. how do we add data plans?

**Comment by karthiksubraveti** [comment]
@pkapatsos-blinq We can currently add different data plans through json editor
![image](https://user-images.githubusercontent.com/8224854/113058759-69ef1c80-9163-11eb-8f4f-f5c9e6f39de1.png)
![image](https://user-images.githubusercontent.com/8224854/113058775-6d82a380-9163-11eb-8ec6-edf82110d0be.png)

**Reply by pkapatsos-blinq** [reply]
Ok I don't see the key of where the default one is being set now ? I get that we can add but where is the setting now  if it does show here ?
Also our point is we are using APNs and want those settings as the preferred , what is the order on what takes precedence ? 
Is it the default or data plans first , then UE APN then policy etc, 
If we step back very simple use case, we want to set an APN to UL/DL AMBR  rate of 5Mbs and push traffic to see if that works or caps at that rate . 
What is needed  or how do we setup as such for us to accomplish this please simple ask or use case.
 
![image](https://user-images.githubusercontent.com/61711855/113062433-b8131880-9181-11eb-9d0c-366adf360d4a.png)

**Reply by pkapatsos-blinq** [reply]
> @pkapatsos-blinq We can currently add different data plans through json editor
> ![image](https://user-images.githubusercontent.com/8224854/113058759-69ef1c80-9163-11eb-8f4f-f5c9e6f39de1.png)
> ![image](https://user-images.githubusercontent.com/8224854/113058775-6d82a380-9163-11eb-8ec6-edf82110d0be.png)

Can you please share how as the editor via the NMS is not setting or how to add these key via that process, 
Im trying via Swagger GUI also but keep getting EOF errros, 
Here  is this the right Key ?
![image](https://user-images.githubusercontent.com/61711855/113303970-958f1580-92cf-11eb-87e7-b9a85de1d6ee.png)
![image](https://user-images.githubusercontent.com/61711855/113304167-c0796980-92cf-11eb-8861-bb4d77cc3af7.png)

**Comment by ulaskozat** [comment]
@pkapatsos-blinq AGW does not enforce UE AMBR, it enforces APN AMBR. UE AMBR should be enforced by RAN. One can add different data plans but assign only one data plan to a subscriber (which sets UE AMBR).

**Comment by pkapatsos-blinq** [comment]
@ulaskozat , Yes understood on enforcing. MME is setting towards the RAN . Shouldn't it be matching the APN-AMBR for Non GBRs or default ? 
Is there anyway we can see what is being applied in terms of AMBR to the attached UE EPS bearers on the AGW ? even if it in the NMS. If not Ok . 

@karthiksubraveti , Sorry the JSON editor at least via the NMS , will not take the settings for me. 
I tried under cellular item adding a new key item of sub_profile.  However even after saving it and exiting coming back it does not stay.
![image](https://user-images.githubusercontent.com/61711855/113173056-08888580-9217-11eb-9e4d-0e8165c3b73e.png)
exit and bck to edit 
![image](https://user-images.githubusercontent.com/61711855/113173276-37066080-9217-11eb-8d29-a83018845c72.png)
 
Can you  please share how do we add or set additional DATA plans via JSON or any other way. Note that I need the actual keys and process on how please am not a developer.

This a bit confusing and my understanding here . The data plan which if I understand is the UE-AMBR is being set  for nonGBR sessions not the APN-AMBR. this means we have be able to make the data plans easily configurable like the APNs to be added via the GUI,
Thanks

**Reply by ulaskozat** [reply]
@pkapatsos-blinq default data plan for UE is hardwired as 200 and 100 mbps. So if this default is not preferred, you need to define new sub_profiles in JSON as @karthiksubraveti has posted, e.g., add something like:
```
"sub_profiles": {
   "gold_plan": {
        "max_dl_bit_rate": 10000000
        "max_ul_bit_rate": 5000000
   } 
}
```

Then, when adding the subscriber, you can specify its data plan as `gold_plan`.

**Reply by ulaskozat** [reply]
I am not sure what the behavior would be if someone uses "default" as the key for a new sub_profile. @karthiksubraveti would that overwrite the default?

**Comment by ulaskozat** [comment, marked as answer]
@pkapatsos-blinq for hardwired values on s1ap, I raised the following PR: https://github.com/magma/magma/pull/5886. You can see a particular pcap for subscriber data plan of (200mbps,100mbps), APN AMBR (1.5mbps,96kbps), dedicated bearer (QCI 1, gbr: 64kbps, 48kbps, mbr: 168kbps, 150kbps) here:

[dump.pcap.zip](https://github.com/magma/magma/files/6247138/dump.pcap.zip)

**Reply by pkapatsos-blinq** [reply]
I did get this answered thanks and  working
